rules/declaration-block-properties-flexible-order/index.js

```diff
@@ -0,0 +1,89 @@
+'use strict';
+
+const stylelint = require('stylelint');
+const _ = require('lodash');
+const utils = require('../../utils');
+const declarationBlockPropertiesSpecifiedOrder = require('../declaration-block-properties-specified-order');
+
+const ruleName = utils.namespace('declaration-block-properties-flexible-order');
+
+const messages = stylelint.utils.ruleMessages(ruleName, {
+	expected: (first, second) => `Expected ${first} to come before ${second}`,
+});
+
+function createExpectedOrder(input) {
```

Comment:
No, it shouldn't. `createExpectedOrder` don't verifies config. It takes valid config as a parameter. An array of string is a valid config for both rules, but `flexible-order` accept different kind of configs also.

`createExpectedOrder` that you have in `flexible-order` will work correctly for both rules.